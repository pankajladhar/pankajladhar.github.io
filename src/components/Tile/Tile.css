.Tile {
  width: 31%;
  border: 1px solid rgba(0, 0, 0, 0.1) !important;
  border-radius: 2px;
  box-shadow: 0 1px 2px rgba(0, 0, 0, 0.07);
  box-sizing: content-box;
  margin: 0px 0 25px 25px;
  position: relative; }
  .Tile .Tile-Image {
    max-height: 190px;
    overflow: hidden; }
    .Tile .Tile-Image img {
      width: 100%;
      display: block; }
  .Tile .Tile-Details {
    padding: 20px; }
    .Tile .Tile-Details .divider {
      margin: 0 70px 20px;
      border: none;
      border-bottom: 1px #e5e5e5 solid; }
  .Tile .ProjectInfo {
    padding-bottom: 50px; }
    .Tile .ProjectInfo .ProjectInfo-Title {
      padding-bottom: 15px;
      color: rgba(0, 0, 0, 0.84);
      font-size: 21px;
      line-height: 24px;
      font-weight: 600;
      font-family: 'Barlow Semi Condensed', sans-serif; }
    .Tile .ProjectInfo .ProjectInfo-Description {
      color: rgba(0, 0, 0, 0.54);
      font-size: 14px;
      line-height: 18px;
      margin: 0; }
  .Tile .AuthorInfo {
    font-size: 12px;
    display: flex;
    align-items: center;
    padding-top: 20px; }
    .Tile .AuthorInfo a {
      color: rgba(0, 0, 0, 0.84);
      display: block; }
      .Tile .AuthorInfo a:hover {
        text-decoration: underline; }
    .Tile .AuthorInfo .AuthorInfo-Avatar {
      width: 40px;
      height: 40px;
      border-radius: 100%;
      border: 2px #4caf4f solid;
      margin-right: 20px; }
    .Tile .AuthorInfo .AuthorInfo-submissionDate {
      color: rgba(0, 0, 0, 0.54);
      padding-top: 5px;
      display: block; }
  .Tile .Tile-CTA {
    background: #f8f8f8;
    border-top: 1px solid rgba(0, 0, 0, 0.1) !important;
    position: absolute;
    bottom: 0;
    width: 100%;
    right: 0;
    padding: 10px 0;
    justify-content: flex-end;
    display: flex; }
    .Tile .Tile-CTA a {
      cursor: pointer;
      font-size: 14px;
      font-weight: normal;
      display: inline-block;
      color: #777;
      text-decoration: none;
      margin-right: 20px; }
      .Tile .Tile-CTA a:hover {
        color: #232222;
        text-decoration: underline; }
      .Tile .Tile-CTA a:focus {
        outline: none; }
