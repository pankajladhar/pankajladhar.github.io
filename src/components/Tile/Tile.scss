.Tile {
  width: 31%;
  border: 1px solid rgba(0,0,0,.1)!important;
  border-radius: 2px;
  box-shadow: 0 1px 2px rgba(0,0,0,0.07);
  box-sizing: content-box;
  margin: 0px 0 25px 25px;
  position: relative;

  .Tile-Image {
    max-height: 190px;
    overflow: hidden;

    img {
      width: 100%;
      display: block;
    }
  }

  .Tile-Details {
    padding: 20px;

    .divider {
      margin: 0 70px 20px;
      border: none;
      border-bottom: 1px #e5e5e5 solid;
    }
  }

  .ProjectInfo {
    padding-bottom: 50px;
    .ProjectInfo-Title {
      padding-bottom: 15px;
      color: rgba(0, 0, 0, .84);
      font-size: 21px;
      line-height: 24px;
      font-weight: 600;
      font-family: 'Barlow Semi Condensed', sans-serif;
    }

    .ProjectInfo-Description {
      color: rgba(0, 0, 0, .54);
      font-size: 14px;
      line-height: 18px;
      margin: 0; 
    }
  }

  .AuthorInfo {
    font-size: 12px;
    display: flex;
    align-items: center;
    padding-top: 20px;
    a {
      color: rgba(0, 0, 0, .84);
      display: block;
      &:hover {
        text-decoration: underline;
      }
    }

    .AuthorInfo-Avatar {
      width: 40px;
      height: 40px;
      border-radius: 100%;
      border: 2px #4caf4f solid;
      margin-right: 20px;
    }

    .AuthorInfo-submissionDate {
      color: rgba(0, 0, 0, .54);
      padding-top: 5px;
      display: block;
    }
  }
  .Tile-CTA {
    background: #f8f8f8;
    border-top: 1px solid rgba(0,0,0,.1)!important;
    position: absolute;
    bottom: 0;
    width: 100%;
    right: 0;
    padding: 10px 0;
    justify-content: flex-end;
    display: flex;
    a {
      cursor: pointer;
      font-size: 14px;
      font-weight: normal;
      display: inline-block;
      color: #777;
      text-decoration: none;
      margin-right: 20px;
      &:hover {
        color: #232222;
        text-decoration: underline;
      }
      &:focus {
        outline: none;
      }
    }
  }
}