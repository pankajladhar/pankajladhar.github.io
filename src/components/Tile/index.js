import React, { Component } from 'react';
//import PropTypes from 'prop-types';
//import classnames from 'classnames';
import './Tile.css';

const propTypes = {}
const defaultProps = {}

export default class Tile extends Component {
    render() {
        return (
            <div className="Tile">
                <div className="Tile-Image">
                    <img src={this.props.imageUrl}  alt={this.props.name}/>
                </div>
                <div className="Tile-Details">
                    <hr className="divider" />
                    <div className="ProjectInfo">
                        <h3 className="ProjectInfo-Title"> {this.props.name} </h3>
                        <p className="ProjectInfo-Description">{this.props.description}</p>
                    </div>
                </div>
                <div className="Tile-CTA">
                    <a href={this.props.demoUrl} target="_blank">Demo</a>
                    <a href={this.props.githubUrl} target="_blank">Code</a>
                </div>
            </div>
        );
    }
}

Tile.propTypes = propTypes;
Tile.defaultProps = defaultProps;
