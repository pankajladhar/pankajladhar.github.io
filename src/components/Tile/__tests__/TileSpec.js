import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';
import Tile from './../index';

describe('Tile', ()=>{
    it('renders correctly', ()=>{
        const tileProps = {
            imageUrl: "imageUrl",
            name: "name",
            description: "description",
            demoUrl: "demoUrl",
            githubUrl: "githubUrl"
        }
        let tree = renderer.create(
            <Tile {...tileProps} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
