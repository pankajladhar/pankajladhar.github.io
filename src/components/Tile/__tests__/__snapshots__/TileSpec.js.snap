// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile renders correctly 1`] = `
<div
  className="Tile"
>
  <div
    className="Tile-Image"
  >
    <a
      href="demoUrl"
      target="_blank"
    >
      <img
        alt="name"
        src="imageUrl"
      />
    </a>
  </div>
  <div
    className="Tile-Details"
  >
    <hr
      className="divider"
    />
    <div
      className="ProjectInfo"
    >
      <h3
        className="ProjectInfo-Title"
      >
         
        name
         
      </h3>
      <p
        className="ProjectInfo-Description"
      >
        description
      </p>
    </div>
  </div>
  <div
    className="Tile-CTA"
  >
    <a
      href="demoUrl"
      target="_blank"
    >
      Demo
    </a>
    <a
      href="githubUrl"
      target="_blank"
    >
      Code
    </a>
  </div>
</div>
`;
